presto-main/src/main/java/com/facebook/presto/type/ArrayType.java

```diff
@@ -65,7 +65,26 @@ public boolean isOrderable()
     @Override
     public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
     {
-        return compareTo(leftBlock, leftPosition, rightBlock, rightPosition) == 0;
+        Block leftArray = leftBlock.getObject(leftPosition, Block.class);
+        Block rightArray = rightBlock.getObject(rightPosition, Block.class);
+
+        if (leftArray.getPositionCount() != rightArray.getPositionCount()) {
+            return false;
+        }
+
+        int len = Math.min(leftArray.getPositionCount(), rightArray.getPositionCount());
+        int index = 0;
+        while (index < len) {
+            checkElementNotNull(leftArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            checkElementNotNull(rightArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            boolean isEqual = elementType.equalTo(leftArray, index, rightArray, index);
```

Comment:
To be clear, we aim for most readability. In some cases, the code is clearer if there's a variable with a descriptive name (e.g., a complex expression). In this case, the expression is trivial, so, as @haozhun suggested, inlining the usage would make the code slightly easier to follow.